src/Solver/Numerov.cpp

```diff
@@ -0,0 +1,249 @@
+#include "Numerov.h"
+
+static inline double square (double x) { return x*x; }
+
+Numerov::Numerov(Potential potential, int nbox)
+{
+    this->potential        = &potential;
+    this->nbox             = nbox;
+    this->solutionEnergy   = 0;
+    this->probability      = std::vector<double>(nbox+1);
+    this->wavefunction     = std::vector<double>(nbox+1);
+    this->tempWavefunction = std::vector<double>(nbox+1);
+
+    // Fill wavefunction with 1 in order to multiply each wavefunction found on separated potentials
+    std::fill(this->wavefunction.begin(), this->wavefunction.end(), 1.0);
+    std::fill(this->tempWavefunction.begin(), this->tempWavefunction.end(), 0.0);
+    this->tempWavefunction[0] = 0.0;
+    this->tempWavefunction[1] = 0.1;
+}
+
+// Numerov Algorithm solves f''(x) + v(x)f(x) = 0, 
+// by considering
+// \left( 1+ \frac{h^2}{12} v(x+h) \right) f(x+h) = 2 \left( 1 - \frac{5h^2}{12} v(x) \right) f(x) - \left( 1 + \frac{h^2}{12} v(x-h) \right) f(x-h).
+// for the Shroedinger equation v(x) = V(x) - E, where V(x) is the potential and E the eigenenergy
+void Numerov::findWavefunction(double energy, std::vector<double> &potential_values)
+{
+    double x;
+    double c = (2.0 * mass / hbar / hbar) * (dx * dx / 12.0);
+    potential_values.push_back(1.0); // tricky fix
```

Comment:
what's this? it should be completely wrong.
I understand that you are adding a 1.0 at the end of the potential, this is probably because of some mismanagement of vector positions. Be careful that numerov depends at the position -2, -1 respect to the search position. Position 0, 1 are fixed at start. Therefore, you should iterate for a number of steps equal to `nbox-2`. Try to print what values of potential are you actually using where you where having crashes.

Just for fun, what are you making is something:
- asymmetrical (in the case of center in 0, as in all of our tests)
- discontinuous
- discontinuous right at boundary

which is a complete mess physically. What would happen physically if this were possible in reality, is that over time you would accumulate wave function density (particles) at boundary due to tunnelling. Due to the small width and low and sharp potential this would generate a "quantum ratchet" effect that would completely wreck havoc to the thermodynamics of the system.

Reply:
I believe it has something to do with Issue #44